pkg/operator/ceph/cluster/watcher.go

```diff
@@ -130,6 +145,333 @@ func (c *clientCluster) onK8sNode(ctx context.Context, object runtime.Object) bo
+
+	// We only need to create the network fence for any one of rbd/cephFS pv.
+	err = c.fenceRbdImage(ctx, node, cluster, clusterInfo, rbdPVList[0])
+	if err != nil {
+		return pkgerror.Wrapf(err, "failed to fence rbd volumes")
+	}
+
+	// else {
+	// 	err = c.fennceCephFSVolumes(ctx, node, cluster, clusterInfo, listCephFSPV[0])
+	// 	if err != nil {
+	// 		return fmt.Errorf("failed to fence cephfs volumes. %v", err)
+	// 	}
+	// }
+
+	return nil
+}
+
+func getCephVolumesInUse(cluster *cephv1.CephCluster, volumesInUse []v1.UniqueVolumeName) ([]string, []string) {
+	var rbdVolumesInUse, cephFSVolumeInUse []string
+
+	for _, volume := range volumesInUse {
+		volumesInuseRemoveK8sPrefix := strings.TrimPrefix(string(volume), "kubernetes.io/csi/")
+		splitVolumeInUseBased := strings.Split(volumesInuseRemoveK8sPrefix, "^")
+		logger.Infof("volumeInUse after split based on '^' %v", splitVolumeInUseBased)
+
+		if len(splitVolumeInUseBased) == 2 && splitVolumeInUseBased[0] == fmt.Sprintf("%s.rbd.csi.ceph.com", cluster.Namespace) {
+			rbdVolumesInUse = append(rbdVolumesInUse, splitVolumeInUseBased[1])
+
+		}
+		// else if len(splitVolumeInUseBased) == 2 && splitVolumeInUseBased[0] == fmt.Sprintf("%s.cephfs.csi.ceph.com", cluster.Namespace) {
+		// 	cephFSVolumeInUse = append(cephFSVolumeInUse, splitVolumeInUseBased[1])
+		// }
+	}
+	return rbdVolumesInUse, cephFSVolumeInUse
+}
+
+func listRBDAndCephFSPV(listPVs *corev1.PersistentVolumeList, cluster *cephv1.CephCluster, rbdVolumesInUse []string) ([]corev1.PersistentVolume, []corev1.PersistentVolume) {
+	var listRbdPV, listCephFSPV []corev1.PersistentVolume
+
+	for _, pv := range listPVs.Items {
+		if pv.Spec.CSI.Driver == fmt.Sprintf("%s.rbd.csi.ceph.com", cluster.Namespace) {
+			for _, rbdVolume := range rbdVolumesInUse {
+				if pv.Spec.CSI.VolumeHandle == rbdVolume {
+					listRbdPV = append(listRbdPV, pv)
+				}
+			}
+		}
+		// else if pv.Spec.CSI.Driver == fmt.Sprintf("%s.cephfs.csi.ceph.com", cluster.Namespace) {
+		// 	for _, cephFSVolume := range cephFSVolumeInUse {
+		// 		if pv.Spec.CSI.VolumeHandle == cephFSVolume {
+		// 			listCephFSPV = append(listCephFSPV, pv)
+		// 		}
+		// 	}
+		// }
+	}
+	return listRbdPV, listCephFSPV
+}
+
+func (c *clientCluster) fenceRbdImage(
+	ctx context.Context, node *corev1.Node, cluster *cephv1.CephCluster,
+	clusterInfo *cephclient.ClusterInfo, rbdPV corev1.PersistentVolume) error {
+
+	logger.Infof("rbd PV NAME %v", rbdPV.Spec.CSI.VolumeAttributes)
+	args := []string{"status", fmt.Sprintf("%s/%s", rbdPV.Spec.CSI.VolumeAttributes["pool"], rbdPV.Spec.CSI.VolumeAttributes["imageName"])}
+	cmd := cephclient.NewRBDCommand(c.context, clusterInfo, args)
+	cmd.JsonOutput = true
+
+	buf, err := cmd.Run()
+	if err != nil {
+		return pkgerror.Wrapf(err, "failed to list watchers for pool/imageName %s/%s.", rbdPV.Spec.CSI.VolumeAttributes["pool"], rbdPV.Spec.CSI.VolumeAttributes["imageName"])
+	}
+
+	logger.Infof("rbd status to get the client ips %v", string(buf))
```

Comment:
debug level?

Reply:
I think it will be better to see in the logs too what IPs we are blocking and it will not be verbose since we already have multiple checks to fall back if fencing is not enabled or other cases.

Reply:
This will currently be printed for every PV on the node? There could be a lot of the PVs.

Reply:
actually, missed adding break if err !=nil. So if `fenceRbdImage` is success it will come out